src/proposer/block_builder.cpp

```diff
@@ -134,9 +135,16 @@ class BlockBuilderImpl : public BlockBuilder {
     // (which happens after the finite supply limit is reached)
     // then there is no reward at all. The reward output will nevertheless
     // be added with an amount of zero.
-    const CScript reward_script = m_settings->reward_destination && reward > 0
-                                      ? GetScriptForDestination(*m_settings->reward_destination)
-                                      : eligible_coin.utxo.GetScriptPubKey();
+    CScript reward_script;
+
+    if (coinbase_script) {
+      reward_script = CScript(coinbase_script->begin(), coinbase_script->end());
```

Comment:
NIT: I believe this can be simplified:
`reward_script = coinbase_script.get();`